### PR TITLE
cli: unify 'contract deploy' and 'contract invoke*'

### LINKED ISSUE #1887: Unify contract deployment and function invocation code
`cli/smartcontract` provides `invokefunction` and `deploy` commands, but `deploy` is effectively a contract invocation now and both commands share a lot of flags, so we can unify their code.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #1887

Depends on #1914

### CODE REVIEW

**Comment by roman-khimov** [comment]
Rebase, please.

File: cli/smartcontract/smart_contract.go [hunk 1]
```diff
@@ -542,73 +529,90 @@ func invokeInternal(ctx *cli.Context, signAndPush bool) error {
 			return cli.NewExitError(err, 1)
 		}
 	}
-
```
Comment (roman-khimov) [inline cli/smartcontract/smart_contract.go:545]: Not really necessary.

File: cli/smartcontract/smart_contract.go [hunk 2]
```diff
@@ -542,73 +529,90 @@ func invokeInternal(ctx *cli.Context, signAndPush bool) error {
...
 		cosignersAccounts, err = cmdargs.GetSignersAccounts(wall, cosigners)
 		if err != nil {
-			return cli.NewExitError(fmt.Errorf("failed to calculate network fee: %w", err), 1)
+			return sender, cli.NewExitError(fmt.Errorf("failed to calculate network fee: %w", err), 1)
 		}
 	}
+
 	gctx, cancel := options.GetTimeoutContext(ctx)
 	defer cancel()
-
```
Comment (roman-khimov) [inline cli/smartcontract/smart_contract.go:565]: ?

**Comment by roman-khimov** [comment]
I also don't really like returning `sender` from `invokeWithArgs`, but looks like it's not easy to avoid this at the moment.

**Comment by AnnaShaleva** [comment]
> I also don't really like returning `sender` from `invokeWithArgs`, but looks like it's not easy to avoid this at the moment.

The other option is to get the sender from context, but this will result in `getAccFromContext`'s code duplication.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**